config-sidecar/script.mjs

```diff
@@ -0,0 +1,124 @@
+    alias: 'u',
+    describe: 'the url to the reload endpoint on geoserver',
+    default: 'http://localhost:8080/geoserver/rest/reload',
+    nargs: 1,
+    string: true,
+  })
+  .option('data-dir-path', { alias: 'D', describe: 'The path to the data dir', nargs: 1, string: true, demandOption: true })
+  .option('update-interval', { alias: 'I', describe: 'The interval between configuration reload', nargs: 1, number: true, default: 120000 })
+  .option('zx-shell', { alias: 'z', describe: 'The shell used by zx', nargs: 1, default: '/bin/bash' })
+  .help('h')
+  .alias('h', 'help').argv;
+
+$.shell = args.zxShell;
+const currentStateFile = join(args.dataDirPath, 'state');
+
+const finished = util.promisify(stream.finished);
+
+const streamToString = (stream) => {
+  return new Promise((resolve, reject) => {
+    const chunks = [];
+    stream.on('data', (chunk) => chunks.push(chunk));
+    stream.on('error', reject);
+    stream.on('end', () => resolve(Buffer.concat(chunks).toString('utf8')));
+  });
+};
+
+const streamToFile = async (stream, path) => {
+  const file = fs.createWriteStream(path);
+
+  for await (const chunk of stream) {
+    if (!file.write(chunk)) {
+      await once(file, 'drain');
+    }
+  }
+
+  file.end();
+
+  await finished(file);
+};
+
+const s3Client = new S3Client({
+  signatureVersion: 'v4',
+  endpoint: args.s3EndpointUrl,
+  forcePathStyle: true,
+  region: REGION,
+});
+
+const updateConfiguration = async () => {
+  const remoteStateStream = await s3Client.send(new GetObjectCommand({ Bucket: args.s3BucketName, Key: args.s3StateKey }));
+  const remoteState = await streamToString(remoteStateStream.Body);
+
+  if (!args.isInitMode && remoteState == (await $`cat ${currentStateFile}`)) {
+    console.log('state unchanged');
+    return;
+  }
+
+  const datadirStream = await s3Client.send(new GetObjectCommand({ Bucket: args.s3BucketName, Key: args.s3DataKey }));
+
+  await streamToFile(datadirStream.Body, DATA_FILE);
+
+  console.log(chalk.blue('unpacking the data dir'));
+  await $`tar -zxf ${DATA_FILE}`;
+
+  await $`cp -r data_dir/* ${args.dataDirPath}`;
+
+  await $`rm -rf ${DATA_FILE}`;
+
+  await $`rm -rf data_dir/*`;
+
+  if (!args.isInitMode) {
+    console.log(chalk.blue('sending config reload request to geoserver'));
+    const response = await fetch(args.geoserverUrl, {
+      method: 'post',
+      headers: {
+        Authorization: 'Basic ' + Buffer.from(`${args.geoserverUser}:${args.geoserverPass}`).toString('base64'),
+      },
+    });
+
+    if (!response.ok) {
+      console.log(chalk.red(`reloading geoserver configuration failed - ${response.statusText}`));
+      process.exit(1);
+    }
+  }
+
+  await fs.writeFile(currentStateFile, remoteState);
+
+  console.log(`State updated to ${remoteState}`);
+};
+
+if (args.isInitMode) {
+  await updateConfiguration();
+  console.log('finished updating, exiting gracefully');
+} else {
+  // wait for geoserver to be ready
+  await sleep(30000);
```

Comment:
30000 should not be a magic number and lets make the default value but also configurable

Reply:
done